csrc/scheduler/expr_eval_sched.h

```diff
@@ -0,0 +1,66 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <scheduler/heuristic.h>
+#include <scheduler/registry.h>
+
+namespace nvfuser {
+
+class Fusion;
+class SchedulerRuntimeInfo;
+class HeuristicSummary;
+
+//! ExprEval scheduler represents the case where we allocate outputs directly using EE. No code is generated.
+class ExprEvalScheduler : public SchedulerEntry {
+ public:
+  explicit ExprEvalScheduler(
+      Fusion* fusion,
+      SchedulerRuntimeInfo& runtime_info,
+      HeuristicSummary* data_cache = nullptr);
+
+  //! This scheduler only accepts matmul and linear nodes
+  static bool canScheduleCompileTime(Fusion* fusion);
+
+  static bool canScheduleRunTime(
+      Fusion* fusion,
+      SchedulerRuntimeInfo& runtime_info,
+      HeuristicSummary* data_cache = nullptr);
+
+  constexpr static ScheduleHeuristic heuristicType() {
+    return ScheduleHeuristic::ExprEval;
+  }
+
+  void schedule(Fusion* fusion) override;
+
+ private:
+  void computeHeuristics(
+      Fusion* fusion,
+      SchedulerRuntimeInfo& runtime_info,
+      HeuristicSummary* data_cache = nullptr);
+};
+
+//! Provides a dummy heuristic type to ensure
+//!  unified interface on ExprEval scheduler.
+class ExprEvalHeuristic : public HeuristicParams {
```

Comment:
I'm not sure what the comment here means.

Reply:
Can you just use `HeuristicParams` in place of `ExprEvalHeuristic` or is the problem that that class is pure virtual. You _could_ consider just providing trivial defaults for those virtual methods then you wouldn't need this class.